### PR TITLE
refactor: refactor group e2e test setup

### PR DESCRIPTION
**Pull request by glnro**
Leverages native sdk types for writing policies & proposals to file

### CODE REVIEW

File: tests/e2e/e2e_setup_test.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	b64 "encoding/base64"
 	"encoding/json"
 	"fmt"
+	"github.com/cosmos/cosmos-sdk/x/group"
```
Comment (mmulji-ic) [inline tests/e2e/e2e_setup_test.go]: Should be moved down with the other sdk incl.

File: tests/e2e/e2e_groups_test.go [hunk 2]
```diff
@@ -1,14 +1,14 @@
 package e2e
```
Comment (mmulji-ic) [inline tests/e2e/e2e_groups_test.go:1]: Missing some docs to explain the flows that are being tested

File: tests/e2e/e2e_groups_test.go [hunk 3]
```diff
@@ -162,16 +159,18 @@ func (s *IntegrationTestSuite) prepareGroupFiles(c *chain, adminAddr string, mem
...
 		},
 	}
+
 	s.writeGroupMembers(c, members, originalMembersFilename)
 
-	newMembers := append(members, GroupMember{
+	newMembers := append(members, group.MemberRequest{
 		Address:  member3Address,
 		Weight:   "1",
 		Metadata: "Charlie",
```
Comment (mmulji-ic) [inline tests/e2e/e2e_groups_test.go:188]: Maybe use different meta data from charlieAddr below

File: tests/e2e/e2e_setup_test.go (hunk 1 above)
Comment (yaruwangway) [reply tests/e2e/e2e_setup_test.go]: please sort the imports

File: tests/e2e/e2e_groups_test.go [hunk 4]
```diff
@@ -162,16 +159,18 @@ func (s *IntegrationTestSuite) prepareGroupFiles(c *chain, adminAddr string, mem
...
 	})
+
 	s.writeGroupMembers(c, newMembers, addMemberFilename)
 
-	removeMembers := append(members, GroupMember{
+	removeMembers := append(members, group.MemberRequest{
```
Comment (yaruwangway) [inline tests/e2e/e2e_groups_test.go]: can you just directly change the `weight` , this `append` way remove members require the member to be removed the last member in the slice.

File: tests/e2e/e2e_groups_test.go (hunk 2 above)
Comment (glnro) [reply tests/e2e/e2e_groups_test.go:1]: I'm not sure why we would have docs for a test, the acceptance criteria/tested flows were part of the original ticket for this. There's sufficient logging in this suite to understand what is being tested.

Comment (glnro) [reply tests/e2e/e2e_groups_test.go:1]: There's also a qa document in hub-eng that explains group flows in better detail

File: tests/e2e/e2e_groups_test.go (hunk 3 above)
Comment (glnro) [reply tests/e2e/e2e_groups_test.go:188]: Metadata field in this case is irrelevant as this is test data, changing it has no effect on the test, and in this particular case, we're targeting the same member in both so it makes sense to keep it consistent. However within the scope of this function we should actually be using `member3Address` passed in and not the globally available `charlieAddr`

File: tests/e2e/e2e_groups_test.go (hunk 4 above)
Comment (glnro) [reply tests/e2e/e2e_groups_test.go]: sure thing

File: tests/e2e/e2e_groups_test.go (hunk 2 above)
Comment (mmulji-ic) [reply tests/e2e/e2e_groups_test.go:1]: An overview of whats being tested in comment form, without having a deep dive into the code/read through the logs helps. Remember this is an open source project, other contributors don't have access to hub-eng.

File: tests/e2e/e2e_groups_test.go (hunk 3 above)
Comment (mmulji-ic) [reply tests/e2e/e2e_groups_test.go:188]: I know metadata is not relevant to the test outcome, but I still think its good to keep both test cases clean and not imply connections between, its visually distracting.

Comment (glnro) [reply tests/e2e/e2e_groups_test.go:188]: Right but in this case the metadata is denoting the specific user and is meant to be the same entity...the only thing that's supposed to change is the weight

File: tests/e2e/e2e_groups_test.go (hunk 2 above)
Comment (glnro) [reply tests/e2e/e2e_groups_test.go:1]: Totally fair, let's open an issue then to make sure it's consistent across test suites

**Review by Pantani — APPROVED** [review]
LGMT

**Review by Pantani — APPROVED** [review]
LGMT

**Pull request merged**